### PR TITLE
VirtualMachinePool Documentation

### PR DESCRIPTION
**Pull request by feitnomore**
This PR adds VirtualMachinePool basic details to the KubeVirt user guide.

Signed-off-by: Marcelo Feitoza Parisi <marcelo@feitoza.com.br>

### CODE REVIEW

**Comment by feitnomore** [comment]
Probably it's worth waiting 0.59 release to merge it, as there are some steps that will only work on that version.

**Comment by cwilkers** [comment]
Thanks for the PR @feitnomore !
/ok-to-test

File: docs/virtual_machines/pool.md [hunk 1]
```diff
@@ -0,0 +1,268 @@
+# VirtualMachinePool
+
+A *VirtualMachinePool* tries to ensure that a specified
+number of VirtualMachine replicas and its respectives
```
Comment (cwilkers) [inline docs/virtual_machines/pool.md:4]:
```suggestion
number of VirtualMachine replicas and their respective
```

File: docs/virtual_machines/pool.md [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+VirtualMachineInstances are in ready state at any time.
```
Comment (cwilkers) [inline docs/virtual_machines/pool.md:5]:
```suggestion
VirtualMachineInstances are in the ready state at any time.
```

File: docs/virtual_machines/pool.md [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+In other words, a *VirtualMachinePool* makes sure that a
+VirtualMachine or a set of VirtualMachines is always up
+and ready. 
+
+No state is kept and no guarantees about the maximum number
+of VirtualMachineInstance replicas which are up are given.
```
Comment (cwilkers) [inline docs/virtual_machines/pool.md:11]:
```suggestion
No state is kept and no guarantees are made about the maximum number
of VirtualMachineInstance replicas running at any time.
```

File: docs/virtual_machines/pool.md [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+
+`spec.replicas` can be used to specify how many replicas are wanted. If
+unspecified, the default value is 1. This value can be updated anytime.
+The controller will react to the changes.
+
+`spec.selector` is used by the controller to keep track of managed
+virtual machines. The selector specified there must be able to match the
+virtual machine labels as specified in `spec.virtualMachineTemplate.metadata.labels`.
+If the selector does not match these labels, or they are empty, the
+controller will simply do nothing except from logging an error. The user
```
Comment (cwilkers) [inline docs/virtual_machines/pool.md:34]:
```suggestion
controller will simply do nothing except log an error. The user
```

File: docs/virtual_machines/pool.md [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+is responsible for not creating other virtual machines or
+*VirtualMachinePools* which conflict with the selector and
```
Comment (cwilkers) [inline docs/virtual_machines/pool.md:36]:
```suggestion
is responsible for avoiding the creation of other virtual machines or
*VirtualMachinePools* which may conflict with the selector and
```

File: docs/virtual_machines/pool.md [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+the template labels.
+
+### Creating a VirtualMachinePool
+
+VirtualMachinePool is part of Kubevirt API `pool.kubevirt.io/v1alpha1`.
```
Comment (cwilkers) [inline docs/virtual_machines/pool.md:41]:
```suggestion
VirtualMachinePool is part of the Kubevirt API `pool.kubevirt.io/v1alpha1`.
```

File: docs/virtual_machines/pool.md [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+      targetCPUUtilizationPercentage: 50
+
+
+or use `kubectl autoscale` to define the HPA via the commandline:
+
+    $ kubectl autoscale vmpool vm-pool-cirros --min=3 --max=10 --cpu-percent=50
+
+## Exposing a VirtualMachinePool as a Service
+
+A VirtualMachinePool could be exposed as a service. When
```
Comment (cwilkers) [inline docs/virtual_machines/pool.md:216]:
```suggestion
A VirtualMachinePool may be exposed as a service. When
```

Adjusting tense.

File: docs/virtual_machines/pool.md [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+Objects](http://kubevirt.io/user-guide/virtual_machines/service_objects/)
+for more details.
+
+
+## Using Persistent Storage
+
+> **Note:** DataVolumes are part of [CDI](https://kubevirt.io/user-guide/operations/containerized_data_importer/)
+
+Usage of a `DataVolumeTemplates` within a `spec.virtualMachineTemplate.spec` will result in unique
+persistent storage getting created for each VM within a VMPool. The `DataVolumeTemplate`
```
Comment (cwilkers) [inline docs/virtual_machines/pool.md:249]:
```suggestion
Usage of a `DataVolumeTemplates` within a `spec.virtualMachineTemplate.spec` will result in the creation
of unique persistent storage for each VM within a VMPool. The `DataVolumeTemplate`
```

File: docs/virtual_machines/pool.md [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+name will have the VM's sequential postfix appended to when the VM is created from the
```
Comment (cwilkers) [inline docs/virtual_machines/pool.md:250]:
```suggestion
name will have the VM's sequential postfix appended to it when the VM is created from the
```

File: docs/virtual_machines/pool.md [hunk 10]
```diff
@@ -0,0 +1,268 @@
...
+`spec.virtualMachineTemplate.spec.dataVolumeTemplates` which makes each VM a completely unique stateful workload.
```
Comment (cwilkers) [inline docs/virtual_machines/pool.md:251]:
```suggestion
`spec.virtualMachineTemplate.spec.dataVolumeTemplates`. This makes each VM a completely unique stateful workload.
```

**Review by cwilkers — CHANGES_REQUESTED** [review]
Great feature! I've added a few grammatical corrections to improve readability, feel free to accept or modify as you see fit.

**Comment by feitnomore** [comment]
@cwilkers thanks for helping me on this one. I've accepted the suggested changes.

**Comment by cwilkers** [comment]
Hi @feitnomore , This is a bit of an annoyance, but if you accepted changes through the GitHub interface, then it does not sign off on those commits, which triggers our blocking DCO check.

You can pull the current branch state on your local copy, and squash all your commits, making sure to leave the Signed off by... line intact in the comment.

**Comment by feitnomore** [comment]
Sorry, I think I forgot to request the review again on this one @cwilkers .

**Comment by cwilkers** [comment]
Thanks for the bump, I missed the DCO fix when it went in.

/approve
/lgtm

**Pull request merged**